pkg/proxy/iptables/proxier.go

```diff
@@ -645,6 +646,17 @@ func (proxier *Proxier) OnNodeUpdate(oldNode, node *v1.Node) {
 		return
 	}
 
+	// kube-proxy in LocalModeNodeCIDR mode may cache stale Node.PodCIDR.
+	// It should restart on those situations to reconcile the state.
+	// Ref: https://issues.k8s.io/111321
+	if proxier.localDetector.IsImplemented() && proxier.localDetector.Mode() == proxyconfigapi.LocalModeNodeCIDR {
+		nodeLocalCIDR := proxier.localDetector.IfLocal()[1]
+		if !utilproxy.NodeContainsPodCIDR(node, nodeLocalCIDR) {
+			klog.ErrorS(nil, "Inconsisten statu, kube-proxy configed Local CIDR not present on Node.Spec.PodCIDRs", "node", klog.KObj(node), "local CIDR", nodeLocalCIDR, "PodCIDRs", node.Spec.PodCIDRs)
```

Comment:
spelling?

Make this log message more direct:

"Current Node PodCIDR(s) not longer matches the PodCIDR kube-proxy saw at startup, restarting"